### PR TITLE
updated getusername rule to include getcomputername for malware that checks hostname

### PR DESCRIPTION
**Pull request by emtuls**

### CODE REVIEW

**Comment by williballenthin** [comment]
i like the idea of adding this detection. would you please add it as a new rule, though?

**Review by williballenthin — CHANGES_REQUESTED** [review]
please submit logic as a new rule: `check for sandbox hostname`

File: anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml [hunk 1]
```diff
@@ -15,7 +16,9 @@ rule:
       - ccbf7cba35bab56563c0fbe4237fdc41:0x402B90
   features:
     - and:
-      - api: GetUserName
+      - or:
+        - api: GetUserName
+        - match: get hostname
```
Comment (williballenthin) [inline anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml:21]: add comment here explaining that username often overlaps with hostname in many of these checks

**Review by williballenthin — APPROVED** [review]

File: anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml [hunk 2]
```diff
@@ -15,7 +16,9 @@ rule:
       - ccbf7cba35bab56563c0fbe4237fdc41:0x402B90
   features:
     - and:
-      - api: GetUserName
+      - or:
+        - api: GetUserName
```
Comment (mike-hunhoff) [inline anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml:20]: substitute match for https://github.com/mandiant/capa-rules/blob/master/host-interaction/session/get-session-user-name.yml?

**Comment by emtuls** [comment]
Updated as per both suggestions.

**Review by williballenthin — APPROVED** [review]

**Comment by mr-tz** [comment]
please rename:
```
FAIL: filename doesn't match the rule name: Rename rule file to match the rule name, expected: "check-for-sandbox-username-or-hostname.yml", found: "check-for-sandbox-username-hostname.yml"
```

File: anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml [hunk 3]
```diff
@@ -37,33 +41,37 @@ rule:
...
           description: Trickbot Downloader Username Check
-        - string: /WIN7\-TRAPS/i
+        - string: /WIN7-TRAPS/i
           description: Trickbot Downloader Username Check
         - string: /WDAGUtilityAccount/i
           description: Windows Defender Application Guard
+        - string: /HAL9TH/i
+          description: Windows Defender Emulator
+        - string: /REM/i
+          description: SANS REM VM
```
Comment (mr-tz) [inline anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml:77]:
```suggestion
          description: SANS REM VM

```

Comment (mr-tz) [inline anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml:77]: strings with less than 4 characters won't be extracted, can you add a character or remove this entry?

Comment (emtuls) [reply anti-analysis/anti-vm/vm-detection/check-for-sandbox-username-hostname.yml:77]: Removed.

**Review by mr-tz — APPROVED** [review]
thanks!

**Pull request merged**